File: SDKs/yodlee.com/1.1.0/python/src/sdk/models/shared/providersdataset.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import attribute as shared_attribute
from ..shared import attribute as shared_attribute

class ProvidersDatasetNameEnum(str, Enum):
    BASIC_AGG_DATA = "BASIC_AGG_DATA"
    ADVANCE_AGG_DATA = "ADVANCE_AGG_DATA"
    ACCT_PROFILE = "ACCT_PROFILE"
    DOCUMENT = "DOCUMENT"


@dataclass_json
@dataclasses.dataclass
class ProvidersDatasetInput:
    attribute: Optional[list[shared_attribute.AttributeInput]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attribute') }})
    name: Optional[ProvidersDatasetNameEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    

@dataclass_json
@dataclasses.dataclass
class ProvidersDataset:
    attribute: Optional[list[shared_attribute.Attribute]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attribute') }})
    name: Optional[ProvidersDatasetNameEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    

File: SDKs/yodlee.com/1.1.0/typescript/src/sdk/models/shared/providersdataset.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { AttributeInput } from "./attribute";
import { Attribute } from "./attribute";


export enum ProvidersDatasetNameEnum {
    BasicAggData = "BASIC_AGG_DATA",
    AdvanceAggData = "ADVANCE_AGG_DATA",
    AcctProfile = "ACCT_PROFILE",
    Document = "DOCUMENT"
}


export class ProvidersDatasetInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=attribute", elemType: AttributeInput })
  attribute?: AttributeInput[];

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: ProvidersDatasetNameEnum;
}


export class ProvidersDataset extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=attribute", elemType: Attribute })
  attribute?: Attribute[];

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: ProvidersDatasetNameEnum;
}
